fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateDisassemble.java

```diff
@@ -0,0 +1,151 @@
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite;
+
+import org.apache.doris.analysis.FunctionName;
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.catalog.Function;
+import org.apache.doris.catalog.Function.CompareMode;
+import org.apache.doris.catalog.Type;
+import org.apache.doris.nereids.operators.Operator;
+import org.apache.doris.nereids.operators.plans.logical.LogicalAggregate;
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.functions.AggregateFunction;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.types.DataType;
+
+import com.clearspring.analytics.util.Lists;
+import com.google.common.base.Preconditions;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Used to generate the merge agg node for distributed execution.
+ * Do this in following steps:
+ *  1. clone output expr list, find all agg function
+ *  2. set found agg function intermediaType
+ *  3. create new child plan rooted at new local agg
+ *  4. update the slot referenced by expr of merge agg
+ *  5. create plan rooted at merge agg, return it.
+ */
+public class AggregateDisassemble extends OneRewriteRuleFactory {
+
+    @Override
+    public Rule<Plan> build() {
+        return logicalAggregate().when(p -> {
+            LogicalAggregate logicalAggregation = p.getOperator();
+            return !logicalAggregation.isDisassembled();
+        }).thenApply(ctx -> {
+            Plan plan = ctx.root;
+            Operator operator = plan.getOperator();
+            LogicalAggregate agg = (LogicalAggregate) operator;
+            List<NamedExpression> outputExpressionList = agg.getOutputExpressionList();
+            List<NamedExpression> intermediateAggExpressionList = Lists.newArrayList();
+            for (NamedExpression namedExpression : outputExpressionList) {
+                namedExpression = (NamedExpression) namedExpression.clone();
+                List<AggregateFunction> functionCallList =
+                        namedExpression.collect(org.apache.doris.catalog.AggregateFunction.class::isInstance);
+                for (AggregateFunction functionCall : functionCallList) {
+                    FunctionName functionName = new FunctionName(functionCall.getName());
+                    List<Expression> expressionList = functionCall.getArguments();
+                    List<Type> staleTypeList = expressionList.stream().map(Expression::getDataType)
+                            .map(DataType::toCatalogDataType).collect(Collectors.toList());
+                    Function staleFuncDesc = new Function(functionName, staleTypeList,
```

Comment:
Wrap these 73~83 lines of getting functions from the old framework in a function. At the same time, with TODO, the acquisition and registration of subsequent lines will be modified uniformly.

Reply:
done